DisguiseUnityRenderStream/Runtime/DisguisePresenter.cs

```diff
@@ -0,0 +1,233 @@
+                        options.Add(channel);
+                    }
+
+                    var remoteParameters = FindObjectsByType<DisguiseRemoteParameters>(FindObjectsSortMode.None);
+                    foreach (var sceneParameter in sceneSchema.parameters)
+                    {
+                        var remoteParams = Array.Find(remoteParameters, rp => sceneParameter.key.StartsWith(rp.prefix));
+                        var field = new ObjectField();
+                        field.info = remoteParams.GetMemberInfoFromManagedParameter(sceneParameter);
+
+                        if (field.FieldType == typeof(Texture))
+                        {
+                            options.Add(sceneParameter.displayName);
+                        }
+                    }
+
+                    parameter.options = options.ToArray();
+                }
+            }
+            
+            return managedParameters;
+        }
+#endif
+
+        /// <remarks>
+        /// This class should only be instantiated through <see cref="LoadPrefab"/>.
+        /// </remarks>
+        private DisguisePresenter()
+        {
+            
+        }
+
+        void OnEnable()
+        {
+            DisguiseRenderStream.SceneLoaded += RefreshInput;
+            DisguiseRenderStream.StreamsChanged += RefreshOutput;
+        }
+        
+        void OnDisable()
+        {
+            DisguiseRenderStream.SceneLoaded -= RefreshInput;
+            DisguiseRenderStream.StreamsChanged -= RefreshOutput;
+        }
+
+        void Update()
+        {
+            Assert.IsNotNull(m_OutputPresenter);
+            Assert.IsNotNull(m_InputPresenter);
+
+            VirtualIndexToRealIndex(Selected, out var outputIdx, out var inputIdx);
+            
+            // No output or input selected
+            if (outputIdx < 0 && inputIdx < 0)
+            {
+                m_OutputPresenter.enabled = m_InputPresenter.enabled = false;
+                return;
+            }
+
+            m_OutputPresenter.enabled = outputIdx >= 0;
+            m_InputPresenter.enabled = inputIdx >= 0;
+            
+            m_OutputPresenter.strategy = m_InputPresenter.strategy = PresenterStrategyToBlitStrategy(m_ResizeStrategy);
+
+            if (m_OutputPresenter.enabled)
+                m_OutputPresenter.cameraCapture = m_Outputs[outputIdx];
+            
+            if (m_InputPresenter.enabled)
+                m_InputPresenter.source = m_Inputs[inputIdx];
+        }
+
+        /// <summary>
+        /// Maps a virtual index like <see cref="Selected"/> to real indexes into <see cref="m_Outputs"/> and <see cref="m_Inputs"/>.
+        /// The virtual list is described in <see cref="GetManagedRemoteParameters"/>.
+        /// </summary>
+        /// <remarks>
+        /// <paramref name="outputIdx"/> and <paramref name="inputIdx"/> are -1 when invalid.
+        /// </remarks>
+        void VirtualIndexToRealIndex(int virtualIdx, out int outputIdx, out int inputIdx)
```

Comment:
A (very) inconsequential case here, but in general, instead of using magic numbers, prefer an alternative such as returning  a `bool` or using nullables. This basically forces the user to deal with the "invalid" case (instead of accidentally passing -1 into an array). e.g.
```
(int? outputIdx, int? inputIdx) ToRealIndices(int virtualIdx)
```
Also, it might be more readable if this was split into 2 methods, since the 2 indices are calculated independently as far as I can tell.